workflow/workflow_test.go

```diff
@@ -1194,9 +1196,12 @@ outputs:
 
 func TestInputCancelledStepWorkflow(t *testing.T) {
 	// Run a workflow where the step is cancelled by a value in the input.
-	// This causes a cancellation during deployment. This is also unique
-	// because that cancelled step is the only step, so its cancellation
-	// must be handled properly to prevent a deadlock.
+	// Therefore, the step and stop_if have their dependencies resolved already.
+	// This causes a cancellation that is executed around the time that
+	// the step starts, which could be during deployment. In this test
+	// the cancellation is enforced to be during deployment with delays.
```

Comment:
I'm still stuck on the phrase "cancelled by a value in the input":  the value in the input has no active role -- it's the `stop_if` which causes the step to be cancelled -- I imagine that the test would work just as well if the `stop_if` referenced a literal, so the input has very little to do with it.  The key feature is that the `stop_if` condition can be resolved as soon as the step starts, and therefore the cancellation can be sent even before deployment completes.

I think we want to say something more like,
> Run a workflow in which the step is cancelled immediately after it begins.  In this scenario, the dependencies of the `stop_if` are already resolved when the step begins to be processed which results in the cancellation occurring during deployment.  This test configures the deployer with a delay to allow the cancellation to be delivered before the deployment finishes.

(I don't think that delays actually "enforce" anything, because, in order to work, they are counting on other actions _not_ being delayed, which is only "mostly reliable".)